src/Fixer/ClassNotation/FinalInternalClassFixer.php

```diff
@@ -177,23 +219,39 @@ private function isClassCandidate(TokensAnalyzer $tokensAnalyzer, Tokens $tokens
             return false; // ignore class; it is abstract or already final
         }
 
-        $docToken = $tokens[$tokens->getPrevNonWhitespace($index)];
+        $attributeDecision = $phpDocDecision = null;
 
-        if (!$docToken->isGivenKind(T_DOC_COMMENT)) {
-            return $this->configuration['consider_absent_docblock_as_internal_class'];
+        if ($this->checkAttributes && $attributeIndex = $tokens->getPrevTokenOfKind($index, [[CT::T_ATTRIBUTE_CLOSE]])) {
+            $attributeDecision = $this->isClassCandidateBasedOnAttribute($tokens, $attributeIndex);
+        }
+
+        if ($commentIndex = $tokens->getPrevTokenOfKind($index, [[T_DOC_COMMENT]])) {
+            $phpDocDecision = $this->isClassCandidateBasedOnPhpDoc($tokens, $commentIndex);
+        }
+
+        if (false === $attributeDecision || false === $phpDocDecision) {
+            return false;
+        }
+
+        if (true === $attributeDecision || true === $phpDocDecision) {
+            return true;
         }
 
-        $doc = new DocBlock($docToken->getContent());
+        return $this->configuration['consider_absent_docblock_as_internal_class'];
```

Comment:
If I'm not mistaken, there's no possibility to reach this `return` - if both `$attributeDecision` and `$phpDocDecision` are `false`, `isClassCandidate()` will return `false`; if one of them is `true`, `isClassCandidate()` will return `false` anyway, if both are `true` then `isClassCandidate()` will return `true`. There's no path for going further than both conditions above 🤔.

Reply:
Both `$attributeDecision` and `$phpDocDecision` can be null (their initial value) if no decision was given, which will trigger this case

Reply:
Ah OK, I looked at those methods' return types and they're `bool`, but you're right variables are overridden only in specific circumstances so they may be `null` 👍.